src/Dungeon/Editor.elm

```diff
@@ -83,18 +82,19 @@ generateCandidate model =
         (newCandidate model
             |> Random.map DungeonGenerator.clean
         )
-            `andThen` (\dungeonModel ->
-                        let
-                            _ =
-                                Debug.log "Editor.generateCandidate"
-                                    { length = List.length dungeonModel.rooms
-                                    }
-                        in
-                            if fitness dungeonModel then
-                                constant dungeonModel
-                            else
-                                generateCandidate model
-                      )
+            |> andThen
+                (\dungeonModel ->
+                    let
+                        _ =
+                            Debug.log "Editor.generateCandidate"
```

Comment:
What is `Debug.log`'s type? Oh... weird... it's a magic function! http://package.elm-lang.org/packages/elm-lang/core/latest/Debug 

> Notice that log is not a pure function! It should only be used for investigating bugs or performance problems.

Haha "notice" :)